### PR TITLE
feat:remove progressContainer from DevFature

### PR DESCRIPTION
**Pull request by tejaskh3**
Date: 16 feb, 2024
Developer Name: Tejas

---

## Issue Ticket Number
#1107 

## Description
- In this issue, there is a sliding Progress Bar so that the super user and the assignee user can update his/her task, this task was under the dev flag which needs to be brought into production.

In this feature firstly the process has been removed from DevFeature and now every person can view the progress of any task, on top of that, the Superuser and assignee user
### Documentation Updated?

- [ ] Yes
- [x] No

### Under Feature Flag

- [x] Yes
- [ ] No

### Database Changes

- [ ] Yes
- [x] No

### Breaking Changes

- [ ] Yes
- [x] No

### Development Tested?

- [x] Yes
- [ ] No

## Screenshots
In the attached video, we can see that the feature is now working even outside the staging in my local environment. As the super user updates the progress, another user can view those changes when I switch browsers.

https://github.com/Real-Dev-Squad/website-status/assets/98630752/1cc12bdf-b133-491e-9d7a-98491955088a

</details>

- Fixed the breaking UI
image

## Test Coverage
| Metric          | Value      |
|-----------------|------------|
| Total Test Suites | 8          |
| Test Suites Passed | 8        |
| Total Tests     | 104        |
| Tests Passed    | 104        |
| Total Snapshots | 0          |
| Execution Time  | 21.274s    |

<summary>Screenshot 1</summary>

</details>

### CODE REVIEW

**Review by iamitprakash — CHANGES_REQUESTED** [review]
at 46 sec UI is breaking please fix this, as for me this seems minor change

File: src/components/tasks/card/card.module.scss [hunk 1]
```diff
@@ -142,7 +142,7 @@
 }
 
 .progressContainerUpdated {
-    width: 12rem;
+    width: 18rem;
```
Comment (iamitprakash) [inline src/components/tasks/card/card.module.scss:145]: does this hard coded value won't affect in other screen size

**Review by iamitprakash — CHANGES_REQUESTED** [review]

File: src/components/tasks/card/card.module.scss (hunk 1 above)
Comment (tejaskh3) [reply src/components/tasks/card/card.module.scss:145]: It shouldn't I have checked. But let me check again.

Comment (tejaskh3) [reply src/components/tasks/card/card.module.scss:145]: Removed these, still working fine.

**Comment by satyam73** [comment]
In staging site as a super user I am seeing this screen
image
> View of https://staging-status.realdevsquad.com 

But in your changes I am seeing something like this, is it intended
image
> View of your changes

**Comment by satyam73** [comment]
image
> View of https://staging-status.realdevsquad.com/

image
> View of your changes(items are not getting wrapped)

**Comment by satyam73** [comment]
Please have some styling so that title and value looks separate
image
> View of your changes

See in prod site we can easily distinguish what's the title and what's the value
image
> View of https://staging-status.realdevsquad.com/

**Comment by tejaskh3** [comment]
> In staging site as a super user I am seeing this screen image
> 
> > View of https://staging-status.realdevsquad.com
> 
> But in your changes I am seeing something like this, is it intended image
> 
> > View of your changes

No this was not intentional, this was an mistake. Thankyou for pointing out. This is fixed now please check.
image

**Comment by tejaskh3** [comment]
> Please have some styling so that title and value looks separate image
> 
> > View of your changes
> 
> See in prod site we can easily distinguish what's the title and what's the value image
> 
> > View of https://staging-status.realdevsquad.com/

this is also fixed now please check this out.
image

**Comment by tejaskh3** [comment]
> Please have some styling so that title and value looks separate image
> 
> > View of your changes
> 
> See in prod site we can easily distinguish what's the title and what's the value image
> 
> > View of https://staging-status.realdevsquad.com/

this has been fixed now please check.
image

**Review by shreya-mishra — APPROVED** [review]
LGTM

**Review by ardourApeX — DISMISSED** [review]

**Comment by tejaskh3** [comment]
> > at 46 sec UI is breaking please fix this, as for me this seems minor change
> 
> Okay, will get back to you after working upon this.

I have updated this, please check now.

**Review by Achintya-Chatterjee — CHANGES_REQUESTED** [review]
You should see other people's test stats before giving or uploading it on the PR details

File: src/components/tasks/card/card.module.scss [hunk 2]
```diff
@@ -136,17 +136,23 @@
 .progressContainer {
     width: 12rem;
     display: flex;
-    flex-direction: column;
+    flex-direction: row;
     align-items: center;
     padding-top: 0.5rem;
+    gap: 0.5rem;
```
Comment (Achintya-Chatterjee) [inline src/components/tasks/card/card.module.scss:142]: @tejaskh3 this file has merge conflicts, Please resolve the merge conflicts

File: src/components/tasks/card/progressContainer/index.tsx [hunk 3]
```diff
@@ -82,7 +79,7 @@ const ProgressContainer: FC<ProgressContainerProps> = ({ content }) => {
             content.assignee === userData?.username ||
             !!userData?.roles.super_user
         ) {
-            setIsProgressMade(true);
+            setIsProgressMade((prev) => !prev);
```
Comment (Achintya-Chatterjee) [inline src/components/tasks/card/progressContainer/index.tsx:82]: @tejaskh3 this file also has merge conflicts, Please resolve this

**Review by Achintya-Chatterjee — CHANGES_REQUESTED** [review]

**Comment by tejaskh3** [comment]
Closing this PR as the changes of this PR is handled in #1163

**Pull request closed without merging**